clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -60,10 +62,11 @@
   private final OutputStream mUnderStorageOutputStream;
   private final long mNonce;
   private String mUfsPath;
+  private FileWriteLocationPolicy mLocationPolicy;
+  private final String mHostname;
```

Comment:
If hostname is specified in `OutStreamOptions`, then location policy is NOT used but always writes to that host.

We could also remove hostname from `OutStreamOptions`, add a new policy that simply returns the host specified in the file policy options.  What do you think?